packages/compiler/src/core/types.ts

```diff
@@ -72,44 +87,53 @@ export type TemplatedType = Model | Operation | Interface | Union | Scalar;
 
 export interface TypeMapper {
   partial: boolean;
-  getMappedType(type: TemplateParameter): Type;
-  args: readonly Type[];
-  /** @internal */ map: Map<TemplateParameter, Type>;
+  getMappedType(type: TemplateParameter): Type | Value | IndeterminateEntity;
+  args: readonly (Type | Value | IndeterminateEntity)[];
```

Comment:
@timotheeguerin that's interesting, do we take this change as breaking change?

Our code would fail compile due to this change.

Reply:
widen the union in output is a breaking, in this case that union type is being used by us, it's an output, therefore it's a breaking. as inspired by the point 6 of @xirzec 's summary of what should be considered as breaking https://github.com/Azure/azure-sdk-for-js/issues/22983#issuecomment-1230916210